fix: FIT-159: Storage error logs modal not optimized for dark mode

ricardoantoniocm opened the pull request:
### Reason for change

The Storage error logs modal on the Settings > Cloud Storage page was not optimized for Dark Mode, making the text unreadable when a failed sync occurred. This PR addresses the issue by implementing design improvements to ensure the modal's readability and overall user experience in Dark Mode.

This PR also includes the following improvements:
- Proper handling of the error overflowing the code block container.
- Increase discoverability of Sync Log modal by adding the text "(View Logs)" when displaying the Failed status
- Custom scrollbars
- Added variants to Code Block component to support for negative and warning types.
- Refactored the Code Block to apply the monospace font using the _pre_ tag.

### Screenshots

#### Before:

LSE
<img width="384" alt="image" src="https://github.com/user-attachments/assets/81735490-4b48-4986-afe6-d9e728fd11cb" />
<img width="398" alt="image" src="https://github.com/user-attachments/assets/e703f17b-9c92-42ac-b812-b7eae0ab56be" />

<img width="746" alt="image" src="https://github.com/user-attachments/assets/9b435941-cb7d-4dff-a5dc-af372970793e" />
<img width="730" alt="image" src="https://github.com/user-attachments/assets/18aaa1be-d690-4e25-9a0c-b6f7e1206b47" />

#### After:

LSO
<img width="365" alt="image" src="https://github.com/user-attachments/assets/6b54c975-79ce-413d-b51f-bdb36d06accb" />
<img width="376" alt="image" src="https://github.com/user-attachments/assets/4a7485f2-db6b-48ed-839c-1d9d920f8e64" />

<img width="730" alt="image" src="https://github.com/user-attachments/assets/e98dbe62-f049-40fe-a259-8454f8bfbf8b" />
<img width="732" alt="image" src="https://github.com/user-attachments/assets/b6439504-b653-4da0-bc36-f8cf725f8d69" />

LSE
<img width="376" alt="image" src="https://github.com/user-attachments/assets/e61afc6a-98e5-4479-ae0f-36522df82ca1" />
<img width="368" alt="image" src="https://github.com/user-attachments/assets/acdc2a8f-4456-4d69-b930-805a5ca690f5" />

<img width="738" alt="image" src="https://github.com/user-attachments/assets/7687f682-3c7c-4fda-8fca-c8c71c21b189" />
<img width="744" alt="image" src="https://github.com/user-attachments/assets/5a2774c9-1fa2-4f46-9b6e-36bd142c9e75" />

### Testing

The changes have been verified by:

* **Manually testing in Dark Mode:**
    * Navigating to Settings > Cloud Storage with Dark Mode enabled.
    * Triggering a failed sync to open the Storage error logs modal.
    * Confirming that the text within the modal is now optimized and readable in Dark Mode.
* **Testing copy-to-clipboard functionality:**
    * Verifying that the "Copy" button correctly copies the error log text to the clipboard.
    * Confirming the button label changes to "Copied!" after a successful copy.
* **Checking updated link and text:**
    * Ensuring the documentation link in the modal footer points to the correct troubleshooting page.
    * Verifying the updated text for the "Failed (View Logs)" status.


<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->

ricardoantoniocm committed:
fix: FIT-159: Storage error logs modal is not optimized for Dark mode

ricardoantoniocm committed:
Merge branch 'develop' of https://github.com/heartexlabs/label-studio into fb-fit-159/storage-error-logs-modal-not-optimized-for-dark-mode

ricardoantoniocm commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/15929761217)
> Successfully merged: Already up to date.

ricardoantoniocm commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/15932578465)
> Successfully merged:  9 files changed, 41 insertions(+), 8 deletions(-)

ricardoantoniocm committed:
Merge branch 'develop' into 'fb-fit-159/storage-error-logs-modal-not-optimized-for-dark-mode'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/15932578465

ricardoantoniocm commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/15934890694)
> Successfully merged:  2 files changed, 3 insertions(+), 3 deletions(-)

ricardoantoniocm committed:
Merge branch 'develop' into 'fb-fit-159/storage-error-logs-modal-not-optimized-for-dark-mode'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/15934890694

ricardoantoniocm committed:
Adds support for warning/negative variants to CodeBlock component.

ricardoantoniocm committed:
Improves the style of the error log. Removes footer for whitelabeled environments. Improves look and feel of Card component so the card for added storage is adapted to dark mode. Reuses existing useCopyText utility for copying. Increases the visibility of the Failed status and adds (View Logs) text to convey that it can be clicked.

ricardoantoniocm committed:
Merge https://github.com/heartexlabs/label-studio into fb-fit-159/storage-error-logs-modal-not-optimized-for-dark-mode

ricardoantoniocm committed:
Adds spacing between Sync button and the rest of the card content.

ricardoantoniocm committed:
Uses pre tag to apply monospace font. Removes margin from component.

ricardoantoniocm commented on web/libs/ui/src/lib/code-block/code-block.tsx:
```diff
@@ -1,15 +1,25 @@
+    <div
+      className={`p-3 rounded-small border scrollbar-thin scrollbar-thumb-neutral-border-bold scrollbar-track-transparent ${variantStyles[variant]} ${className || ""}`}
+    >
+      <pre className="whitespace-pre-wrap">{code.trim()}</pre>
```
Monospace font is applied by default to < pre > tag.

ricardoantoniocm commented on web/apps/labelstudio/src/pages/Settings/StorageSettings/StorageSummary.jsx:
```diff
@@ -1,10 +1,11 @@
 import { modal } from "../../../components/Modal/Modal";
 import { Oneof } from "../../../components/Oneof/Oneof";
 import { getLastTraceback } from "../../../utils/helpers";
+import { useCopyText } from "@humansignal/core/lib/hooks/useCopyText";
```
This is implemented as is across the codebase. Will skip for now to avoid scope creep but it is easily doable if it's ok to include with this PR.

bmartel replied:
I would recommend not following this advice, I already downvoted and gave Copilot explicit reasonings as to why. TLDR; the more specific an import path (specifically pertaining to library imports from the app), the better it is able to tree shake. So generally speaking I would err on the side of more specific paths, as even in the case it gave of tight coupling, that is actually more true of the simpler top level library imports.

bmartel commented on web/libs/ui/src/lib/code-block/code-block.tsx:
```diff
@@ -1,15 +1,25 @@
 }) {
+  const variantStyles = {
+    default: "bg-neutral-surface border-neutral-border",
+    warning: "bg-warning-background border-warning-border-subtle",
+    negative: "bg-negative-background border-negative-border-subtle",
+  };
+
   return (
-    <div className={`whitespace-pre-wrap font-mono mt-2 p-3 bg-neutral-surface rounded-md ${className || ""}`}>
-      {code.trim()}
+    <div
+      className={`p-3 rounded-small border scrollbar-thin scrollbar-thumb-neutral-border-bold scrollbar-track-transparent ${variantStyles[variant]} ${className || ""}`}
```
Can we use `cnm` helper here instead of the string concat, as that does not have any ability to properly order, merge or deal with tailwind classes, where the other helper does.

An example is here: https://github.com/HumanSignal/label-studio/blob/5836becb7ad47674f94d16b085308b0f0482125c/web/libs/editor/src/components/TaskSummary/SummaryBadge.tsx#L1

ricardoantoniocm replied:
Thanks! Refactored the code to use the `cnm` helper.

ricardoantoniocm committed:
Refactors to use cnm helper instead of string concat.

bmartel commented on web/apps/labelstudio/src/pages/Settings/StorageSettings/StorageSummary.jsx:
```diff
@@ -37,34 +38,46 @@ export const StorageSummary = ({ target, storage, className, storageTypes = [] }
       `meta = ${JSON.stringify(storage.meta)}\n`;
-    const targetType = target === "export" ? "Target" : "Source";
+
+    // Component to handle copy functionality within the modal
+    const CopyButton = () => {
+      const [copyText, copied] = useCopyText(msg);
+
+      return (
+        <Button
+          icon={<IconFileCopy />}
+          onClick={copyText}
+          disabled={copied}
+          style={{ minWidth: "100px", justifyContent: "center" }}
+        >
+          {copied ? "Copied!" : "Copy"}
+        </Button>
+      );
+    };
```
Components should always be defined at module level, not within another Component. Move this up and out, and use the component props to take the `msg`.

ricardoantoniocm replied:
Applied recommended refactoring to this and the LSE version of StorageSummary.

ricardoantoniocm committed:
Refactors to define CopyButton at module level and to take in the msg as a parameter.

ricardoantoniocm committed:
Merge branch 'develop' of https://github.com/heartexlabs/label-studio into fb-fit-159/storage-error-logs-modal-not-optimized-for-dark-mode

ricardoantoniocm commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/16169909166)

ricardoantoniocm committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16169909166

ricardoantoniocm committed:
Merge branch 'develop' into 'fb-fit-159/storage-error-logs-modal-not-optimized-for-dark-mode'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16169909166

ricardoantoniocm commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/16170681722)
> Successfully merged: Already up to date.

ricardoantoniocm committed:
Merge branch 'develop' of https://github.com/heartexlabs/label-studio into fb-fit-159/storage-error-logs-modal-not-optimized-for-dark-mode

ricardoantoniocm committed:
Improves the layout of the Storage Sync Error Log further.

ricardoantoniocm committed:
Adds spacing between storage summager and storage sync button.

ricardoantoniocm commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/16178107263)
> Successfully merged:  2 files changed, 91 insertions(+), 9 deletions(-)

ricardoantoniocm committed:
Merge branch 'develop' into 'fb-fit-159/storage-error-logs-modal-not-optimized-for-dark-mode'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16178107263
